### PR TITLE
chore: cleanup repo lint + typechecking

### PR DESCRIPTION
**Pull request by bradzacher**
I noticed a lot of missing coverage with our linting and typechecking setup.
- The top-level test and tools directories weren't being linted/typechecked
- `.js` files weren't being linted or formatted around the place
- Some of our `.d.ts` files weren't being linted or formatted

In some cases files were just part of ignore files when they shouldn't've been. 
In other cases they weren't properly covered by the nx command config.

This PR fixes all of that with a small reorganisation of the codebase and addition of some config files. It also upgrades prettier.

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -1,22 +1,16 @@
...
 **/.vs
-packages/eslint-plugin-tslint/tests/test-tslint-rules-directory/alwaysFailRule.js
-packages/eslint-plugin/src/configs/*.json
 CONTRIBUTORS.md
 packages/ast-spec/src/*/*/fixtures/_error_/*/fixture.ts
 
-# prettier doesn't yet support auto-accessors
-packages/ast-spec/src/element/AccessorProperty/fixtures
+# prettier errors on this case because it's semantically valid
+packages/ast-spec/src/element/AccessorProperty/fixtures/modifier-abstract-with-value/fixture.ts
```
Comment (JoshuaKGoldberg) [inline .prettierignore:10]: Maybe we should turn off Prettier for all fixtures?

File: packages/website/src/vendor/typescript-vfs.d.ts [hunk 2]
```diff
@@ -1,21 +1,26 @@
-declare type System = import('typescript').System;
-declare type CompilerOptions = import('typescript').CompilerOptions;
-declare type CustomTransformers = import('typescript').CustomTransformers;
-declare type LanguageServiceHost = import('typescript').LanguageServiceHost;
-declare type CompilerHost = import('typescript').CompilerHost;
-declare type SourceFile = import('typescript').SourceFile;
-declare type TS = typeof import('typescript');
+/* eslint-disable import/no-duplicates -- bug -> rule suggests merging to `import * as ns, {...}` which is invalid TS syntax */
```
Comment (JoshuaKGoldberg) [inline packages/website/src/vendor/typescript-vfs.d.ts:1]: Is this bug filed anywhere on GitHub?

File: package.json [hunk 3]
```diff
@@ -30,19 +30,19 @@
...
+    "generate-sponsors": "tsx ./packages/repo-tools/generate-sponsors.ts",
+    "generate-website-dts": "tsx ./packages/repo-tools/generate-website-dts.ts",
     "generate-lib": "nx generate-lib @typescript-eslint/scope-manager",
-    "lint-fix": "eslint . --fix",
+    "lint-fix": "yarn lint --fix",
     "lint-markdown-fix": "yarn lint-markdown --fix",
     "lint-markdown": "markdownlint \"**/*.md\" --config=.markdownlint.json --ignore-path=.markdownlintignore",
     "lint": "nx run-many --target=lint --parallel",
-    "postinstall": "yarn tsx ./tools/postinstall.ts",
+    "postinstall": "tsx ./packages/repo-tools/postinstall.ts",
```
Comment (JoshuaKGoldberg) [inline package.json:41]:
```suggestion
    "postinstall": "tsx ./packages/repo-tools/src/postinstall.ts",
```

File: packages/integration-tests/jest.config.js [hunk 4]
```diff
@@ -5,10 +5,13 @@ require('ts-node').register({
   transpileOnly: true,
   files: ['./pack-packages.ts'],
 });
-const { tseslintPackages } = require('./pack-packages');
+const { tseslintPackages } = require('./tests/pack-packages');
```
Comment (JoshuaKGoldberg) [inline packages/integration-tests/jest.config.js:8]:
```suggestion
const { tseslintPackages } = require('./tools/pack-packages');
```

Comment (JoshuaKGoldberg) [reply packages/integration-tests/jest.config.js:8]: And then tests are failing:

```plaintext
 FAIL  tests/markdown.test.ts
  ● markdown › should lint successfully
    thrown: Array [
      [Error: ENOENT: no such file or directory, lstat '/Users/josh/repos/typescript-eslint/packages/integration-tests/tools/fixtures/markdown'],
```

Some issues with relative directory naming

File: .eslintrc.js [hunk 5]
```diff
@@ -14,6 +16,9 @@ module.exports = {
     es6: true,
     node: true,
   },
+  globals: {
+    globalThis: true,
```
Comment (JoshuaKGoldberg) [inline .eslintrc.js:20]:
```suggestion
    es2020: true,
    node: true,
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Lots of changes here that look good - but the tests are borked?

File: packages/website/src/vendor/sandbox.d.ts [hunk 6]
```diff
@@ -1,10 +1,10 @@
-import { TypeScriptWorker } from './tsWorker';
```
Comment (armano2) [inline packages/website/src/vendor/sandbox.d.ts:1]: this file is downloaded from ts, we should not modify it.

File: .prettierignore (hunk 1 above)
Comment (bradzacher) [reply .prettierignore:10]: I'd rather we keep everything formatted and special case the things that are broken, personally.

File: packages/website/src/vendor/sandbox.d.ts (hunk 6 above)
Comment (bradzacher) [reply packages/website/src/vendor/sandbox.d.ts:1]: The changes I made:
- switch `any` for `unknown`
- replace inline type import with type-only import declarations
- deduplicate code
- apply formatter

Functionally the code is the ecactly same - it just adheres to our code style now.

Comment (armano2) [reply packages/website/src/vendor/sandbox.d.ts:1]: you should update tools/generate-website-dts.ts, to apply those changes than :)

Comment (bradzacher) [reply packages/website/src/vendor/sandbox.d.ts:1]: I didn't realise they were generated!

**Review by JoshuaKGoldberg — APPROVED** [review]
🔥

**Pull request merged**